service-base/src/main/java/org/eclipse/hono/service/management/credentials/PasswordSecret.java

```diff
@@ -123,6 +123,23 @@ public void checkValidity() {
         }
     }
 
+    /**
+     * Verifies that only the "password-plain" field is set in the Password Secret, i.e. the password is not hashed.
+     *
+     * @throws IllegalStateException if this is not the case.
+     */
+    public void checkIsPlainText() {
+        if (Strings.isNullOrEmpty(passwordPlain)) {
+            throw new IllegalStateException(String.format("'%s' must be provided", RegistryManagementConstants.FIELD_SECRETS_PWD_PLAIN));
+        }
+        if (!Strings.isNullOrEmpty(hashFunction)) {
+            throw new IllegalStateException(String.format("'%s' must be empty", RegistryManagementConstants.FIELD_SECRETS_HASH_FUNCTION));
+        }
+        if (!Strings.isNullOrEmpty(passwordHash)) {
+            throw new IllegalStateException(String.format("'%s' must be empty", RegistryManagementConstants.FIELD_SECRETS_PWD_HASH));
+        }
```

Comment:
what about the *salt*? IMHO we should also not accept any salt provided by a client, right?

Reply:
Updated !